python/cudf/cudf/core/reshape.py

```diff
@@ -367,3 +367,41 @@ def length_check(obj, name):
             df_list.append(col_enc_df)
 
         return concat(df_list, axis=1).drop(labels=columns)
+
+
+def merge_sorted(objs, keys=None, ascending=True, na_position="last"):
```

Comment:
Note: Perhaps we also want to allow the user to specify that the index should be used for sorting (e.g. `index=True`)